lib/packages.js

```diff
@@ -379,6 +379,16 @@ function _sortPackages(cache) {
   const unsorted = Object.keys(cache).map(name => cache[name]);
   const sorted = [];
 
+  const cyclicDetectionThreshold = 10;
```

Comment:
maybe a comment in the code about why 10 was chosen here, and should it be configurable?

Reply:
I believe it's arbitrarily chosen, which means it can cause false positives, which we don't want. We'll need to change the implementation a bit.

It seems like we'd be able to quickly detect when there are no changes to the unsorted list -- indicating that no progress is being made, hence there must be a circular dependency somewhere. We can dump out which unsorted packages remain in the log and people can sort it out from there. This would be the quickest fix, I think.

Reply:
Currently the code is detecting that there are no changes to the unsorted list. The way it's being achieved is that increments a count for how many times a package has been added to the unsorted list. If it has been added to the unsorted list "x" amount of times then I assumed the unsorted list wasn't changing and then logged and error then exited. I can change the implementation to keep track of a "previous list" and a "current list" and if the items are the same then we know we have a cyclic dependency.